atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTable.java

```diff
@@ -134,7 +134,7 @@ public void checkDatabaseVersion() {
     }
 
     @Override
-    public void compactInternally() {
+    public void compactInternally(boolean unused) {
         // VACUUM FULL is /really/ what we want here, but it takes out a table lock
```

Comment:
Interesting comment, suggests we should maybe have a similar set up in postgres.  Don't need to worry about this just now though.

Reply:
Postgres docs: https://www.postgresql.org/docs/9.5/static/sql-vacuum.html

> 
Plain VACUUM (without FULL) simply reclaims space and makes it available for re-use. This form of the command can operate in parallel with normal reading and writing of the table, as an exclusive lock is not obtained. However, extra space is not returned to the operating system (in most cases); it's just kept available for re-use within the same table. VACUUM FULL rewrites the entire contents of the table into a new disk file with no extra space, allowing unused space to be returned to the operating system. This form is much slower and requires an exclusive lock on each table while it is being processed.

So interestingly the (plain) VACUUM we run for postgres seems to be analogous to SHRINK SPACE COMPACT - and we always run the conservative command in postgres, never running VACUUM FULL (which is analogous to the blocking SHRINK SPACE). Could be something we could do in the future, yes.